src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/IntegerDecoder.cs

```diff
@@ -41,11 +42,29 @@ public bool StartDecode(byte b, int prefixLength)
         /// <returns>If the integer has been fully decoded, true. Otherwise, false -- <see cref="Decode(byte)"/> must be called on subsequent bytes.</returns>
         public bool Decode(byte b)
         {
-            _i = _i + (b & 127) * (1 << _m);
+            if (_m - BitOperations.LeadingZeroCount(b) > 0)
```

Comment:
I'm not understanding the logic here, can you explain?

Reply:
Is `b` being implicitly cast to `uint` here? If so, we should make the cast explicit, because that wasn't at all obvious to me.

Reply:
Also, it seems like we're implicitly handling two slightly different cases here, that is, high bit set or high bit clear. Both seem to work as far as I can tell, but they work for slightly different reasons. It would be nice to add a comment explaining this.

Reply:
Also, I think this test would be clearer if it were written as `if (BitOperations.LeadingZeroCount((uint)b) < _m)`. That makes it more obvious to me that a digit will be lost by shifting.

Reply:
Thanks. I've added a comment explaining this.